src/libcore/python/xml_v.cpp

```diff
@@ -83,12 +83,16 @@ std::string get_type(const py::dict &dict) {
 MTS_VARIANT ref<Object> load_dict(const py::dict &dict) {
 
     std::string type = get_type(dict);
-    Properties props(type);
 
-    // TODO unfortunately this has to be done by hand now
-    bool within_emitter = (type == "area") || (type == "constant") ||
-                          (type == "directional") || (type == "envmap") ||
-                          (type == "point");
+    const Class *class_;
+    if (type == "scene")
+        class_ = Class::for_name("Scene", GET_VARIANT());
+    else
+        class_ = PluginManager::instance()->get_plugin_class(type, GET_VARIANT());
+
+    bool within_emitter = (class_->parent()->name() == "Emitter");
```

Comment:
I think you should go up the full hierarchy in case of more inheritance levels.

Reply:
The XML parser doesn't do that.. There must be a reason 🤔